test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -299,6 +300,79 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 			ginkgo.By("delete the waiting 'prod' workload so that it does not get admitted during teardown")
 			gomega.Expect(k8sClient.Delete(ctx, prodWl)).Should(gomega.Succeed())
 		})
-	})
 
+		ginkgo.It("Should move the evicted workload at the end of the queue", func() {
+			localQueueName := "eviction-lq"
+
+			// the workloads are created with a 5 cpu resource requirement to ensure only one can fit at a given time,
+			// letting them all to time out, we should see a circular buffer admission pattern
+			workloads := []*kueue.Workload{
+				testing.MakeWorkload("prod1", ns.Name).Queue(localQueueName).Request(corev1.ResourceCPU, "5").Obj(),
+				testing.MakeWorkload("prod2", ns.Name).Queue(localQueueName).Request(corev1.ResourceCPU, "5").Obj(),
+				testing.MakeWorkload("prod3", ns.Name).Queue(localQueueName).Request(corev1.ResourceCPU, "5").Obj(),
+			}
+
+			ginkgo.By("create the workloads", func() {
+				tkr := time.NewTicker(time.Second)
+				for _, wl := range workloads {
+					ginkgo.By(fmt.Sprintf("create the '%s' workload", wl.Name), func() {
+						gomega.Expect(k8sClient.Create(ctx, wl)).Should(gomega.Succeed())
+						gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), wl)).Should(gomega.Succeed())
+						ginkgo.GinkgoLogr.Info(wl.Name+" created", "at", wl.CreationTimestamp)
+						// since metav1.Time has only second resolution, wait one second to avoid
+						// any potential creation time collision
+						<-tkr.C
+					})
+				}
+			})
+
+			ginkgo.By("create the local queue to start admission", func() {
+				lq := testing.MakeLocalQueue(localQueueName, ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, lq)).Should(gomega.Succeed())
+			})
+
+			getAdmittedWorkloads := func() []int {
+				ret := make([]int, 0, len(workloads))
+				for i, wl := range workloads {
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), wl)).Should(gomega.Succeed())
+					if apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadAdmitted) {
+						ret = append(ret, i)
+					}
+				}
+				return ret
+			}
+
+			ginkgo.By("check admission order", func() {
+				prev := -1
+				gomega.Eventually(func() bool {
+					admitted := getAdmittedWorkloads()
+					if len(admitted) > 0 {
+						prev = admitted[0]
+					}
+					return prev != -1
+				}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+				for i := 0; i < 2*len(workloads); i++ {
```

Comment:
Why do we need this many iterations? I would suggest to unfold the loop and explicitely say in the comments what transition is expected.  I think we should strive to make the test fast, while maintaining the coverage, but it isn't clear how the looped steps increase coverage.

What I would suggest is to make use 3 workloads, the first one is timed out and moves at the end. Then, workloads 2 and 3 complete successfully. Then, workload 1 after requeing completes successfully too. It is important the workload can succeed after requeue, so to show that the timeout is computed from the most recent admission, and not from creation.

Reply:
I would be ok with that if there are good reasons. In particular one question that puzzles me is why the number of iterations is 2*len(workloads). Please elaborate in a comment so that the reasoning it is clear to the future readers too.

Also, how long does the take currently take?

Reply:
2*len(...) ensures that all workloads are queued at the back of the list and be considered for scheduling later on (not lost in the process)

the test takes about 3 sec to create the workloads, and another @2*3*3sec to check the scheduling order

Reply:
So it is roughly `21`s total vs `6-7`s (if we followed the suggestion to wait only for the first to timeout and then complete them all successfully). 

> 2*len(...) ensures that all workloads are queued at the back of the list and be considered for scheduling later on (not lost in the process)

In the scheme described above, we could check that the first workload is requeued properly.

Is there a code path that is explored additionally by repeating the timeout?

Reply:
done